ercc/registry/registry.go

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+Copyright 2020 Intel Corporation
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package registry
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/hyperledger-labs/fabric-private-chaincode/ercc/attestation"
+	"github.com/hyperledger-labs/fabric-private-chaincode/internal/protos"
+	"github.com/hyperledger-labs/fabric-private-chaincode/internal/utils"
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+	"github.com/pkg/errors"
+)
+
+type Contract struct {
+	contractapi.Contract
+
+	Verifier   attestation.VerifierInterface
+	PEvaluator utils.PolicyEvaluatorInterface
+}
+
+// returns a set of credentials registered for a given chaincode id
+// Note: to get the endpoints of FPC endorsing peers do the following:
+// - discover all endorsing peers (and their endpoints) for the FPC chaincode using "normal" lifecycle
+// - query `getEnclaveId` at all the peers discovered
+// - query `queryListEnclaveCredentials` with all received enclave_ids
+// this gives you the endpoints and credentials including enclave_vk, and chaincode_ek
+//
+// Note that this implementation returns a set of serialized credentials in order to send it to the receiver.
+// That is, the receiver needs to deserialize the return value into []Credentials
+// TODO look into custom serializer to make this call more aligned with the FPC API in `interfaces.md`
```

Comment:
well, we didn't really define the encoding in `interfaces.md` (there is only a "maybe" which is anyway different from what we talk now).  From discussion this morning it seems we have agreed on base64-encoding of (standard binary) protobuf encoding of Objects?